src/index.js

```diff
@@ -39,17 +39,35 @@ export default function fetch(url, opts) {
 	return new fetch.Promise((resolve, reject) => {
 		// build request object
 		const request = new Request(url, opts);
+		const signal = getAbortSignal(request);
+		if (signal.aborted) {
+			reject(new FetchError(`Fetch to ${request.url} has been aborted`, 'aborted'));
```

Comment:
any way this could be more inline with the browser spec? rejecting with an `AbortError`? when using with isomorphic fetch this would result in having to catch two types of errors for node and browser

https://dom.spec.whatwg.org/#aborting-ongoing-activities